modules/tensor_mechanics/include/materials/ComputeStressEosBase.h

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTESTRESSEOSBASE_H
+#define COMPUTESTRESSEOSBASE_H
+
+#include "ComputeStressBase.h"
+
+/**
+ * ComputeStressEosBase adds a volumetric extra stress including a Birch-Murnaghan EOS
+ * and bulk viscosity damping for shock propagation
+ * that is substituted to the volumetric stress
+ * calculated by the constitutive model.
+ */
+class ComputeStressEosBase;
```

Comment:
I'd prefer to have `ComputeEquationOfStateStressBase` rather than `Eos` in the class name and in the class description.  The term equation of state also seems too broad for your implementation of Birch-Murnaghan, so perhaps `ComputeBirchMurnaghanEquationOfStress` is better, without the base class designation.